streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -521,6 +522,10 @@ void commit(final boolean startNewTransaction, final Map<TopicPartition, Long> p
             }
         } catch (final CommitFailedException | ProducerFencedException error) {
             throw new TaskMigratedException(this, error);
+        } catch (final RetriableCommitFailedException error) {
+            // commitSync throws this error and can be ignored (since EOS is not enabled, even if the task crashed
+            // immediately after this commit, we would just reprocess those records again)
+            log.info("Committing failed with a non-fatal error, we can ignore this since commit may succeed still");
```

Comment:
I'm actually pretty sure we wouldn't ever get this exception, as we set a `rebalanceInProgress` flag when encoding the subscription, and therefore we shouldn't ever try to commit between the JoinGroup and SyncGroup. But, I think it's best to be safe here and catch it, since as you mentioned it shouldn't be a problem

Reply:
Yes I'm thinking about the same when reviewing your PR and while reading `rebalanceInProgress` but I felt it is safe to catch it and log as an INFO here.

Reply:
Just in case, it's probably a good idea to include the exception in the message.

Reply:
Ack